### DISCUSSION TITLE
Moodle, SEB and SSO getting 403 response

### DISCUSSION THREAD

**Discussion by DevinMonroe** [SafeExamBrowser/seb-win-refactoring#749] [General]
Hello, 

I am posting this there in case this issue is something with SEB while I continue to investigate Moodle and SSO  as being the source of this issue as well.

There is a specific use case where we keep receiving a 403 error inside SEB. The scenario is this: 

1) user logs into the Moodle installation using OAuth/SSO.
2) user navigates to quiz and clicks "Launch Safe Exam Browser" 
3) user is presented with Moodle main login screen
4) user clicks SSO login option and receives a 403 error

In the Moodle logs it shows the error:

> The user with id ‘307’ has been prevented from accessing quiz with id ‘34’ by the Safe Exam Browser access plugin. The reason was ‘Invalid SEB config key’. Expected config key: ‘35abe092c8754fb2faf8128326e28c93f41883288df42d6cd5500bee4f8f9b80’. Received config key: ‘’. Received browser exam key: ‘’.

Things to note: 

This does not happen when not using SSO to login. 

This does not happen if we use the option to "Download configuration" and open the configuration file.

I was not able to see any other errors in the logs that related to this.  

I am not sure if the issue is with Moodle, SSO or SEB.  

It is my understanding that the "Launch Safe Exam Browser" button should work on it's own as long as you are using the correct versions of SEB. I have tried on both Mac and PC and we have tried with SE  3.3 and 3.5 so far. 

Any advice, suggestions or help would be greatly appreciated. Please let me know if any further information needs to be provided or if I should be looking someone else for help.

**Comment by dbuechel** [comment]
Have you tried the latest release version 3.6.0? We have implemented a (potential) fix for this issue:

> - [Fixed missing BEK and CK headers after e.g. authentication redirection.](https://github.com/SafeExamBrowser/seb-win-refactoring/releases/tag/v3.6.0)

**Reply by DevinMonroe** [reply]
No we had not! Thank you for pointing this out the last time I checked I didn't see the new release. I am using a Mac but I will have those with Windows clients test this new version.

Will there be a new release for Mac as well? The issue we are having is on Mac and Windows.

**Reply by dbuechel** [reply]
Excellent, let's hope that indeed fixes the issue. As for the macOS side of things, @danschlet can give you the most accurate information.

**Reply by danschlet** [reply]
You need to create a separate issue in the Mac repository https://github.com/SafeExamBrowser/seb-mac and provide all necessary information as described in the bug report template, e.g. demo login link. Please note that only Moodle 4.1 or newer or the latest 4.x versions use the updated Moodle SEB integration, which allows to use the modern web browser engine in SEB for macOS/iOS. We cannot debug issues which occur with the outdated classic WebView.

**Reply by DevinMonroe** [reply]
@danschlet Thank you!

**Comment by danschlet** [comment]
Btw., the issue definitely will be the particular SSO used, if this happens in the Windows and Mac SEB versions. We test with Moodle without SSO and with our Shibboleth-based SSO and it works.
